digital_land/update.py

```diff
@@ -74,126 +72,68 @@ def has_collected_resource(log_item):
-    if "endpoint" not in resource_entry or "organisation" not in resource_entry:
-        raise ValueError("Missing required fields 'endpoint' or 'organisation'")
-    source_entries = source_register.entries
-    if resource_entry["endpoint"] in source_register.record:
-        for idx in source_register.record[resource_entry["endpoint"]]:
-            if (
-                resource_entry["organisation"]
-                == source_entries[idx].item["organisation"]
-            ):
-                if source_entries[idx].item["end-date"]:
-                    # Entry already exists with an end-date
-                    print(
-                        "WARNING: source end-date {} found for URL {}".format(
-                            source_entries[idx].item["end-date"],
-                            resource_entry["endpoint-url"],
-                        )
-                    )
-                    break
-                else:
-                    # No op if active entry already exists
-                    print(
-                        "Active source entry already exists for organisation {} and URL {}".format(
-                            resource_entry["organisation"],
-                            resource_entry["endpoint-url"],
-                        )
-                    )
-                    return
-
-    source_entry = {
-        "collection": resource_entry["collection"]
-        if "collection" in resource_entry
-        else resource_entry["pipeline"],
-        "pipeline": resource_entry["pipeline"],
-        "organisation": resource_entry["organisation"],
-        "endpoint": resource_entry["endpoint"],
-        "documentation-url": resource_entry["documentation-url"],
-        "licence": resource_entry["licence"],
-        "attribution": resource_entry["attribution"],
-        "entry-date": date.today().strftime("%Y-%m-%d"),
-        "end-date": datetime.strptime(resource_entry["end-date"], "%Y-%m-%d").date()
-        if "end-date" in resource_entry
-        else "",
-    }
+    collection = Collection(collection_directory)
+    collection.load()
+
+    entry["endpoint"] = hash_value(entry["endpoint-url"])
+
+    add_new_endpoint(entry, collection.endpoint)
+    add_new_source(entry, collection.source)
 
-    # If empty start-date set by user then no date should be set. Otherwise set user-specified date or use current date
-    if "start-date" in resource_entry:
-        source_entry["start-date"] = (
-            datetime.strptime(resource_entry["start-date"], "%Y-%m-%d").date()
-            if resource_entry["start-date"]
-            else ""
-        )
-    else:
-        source_entry["start-date"] = date.today().strftime("%Y-%m-%d")
-
-    source_item = Item(source_entry)
-    source_register.add(source_item)
+    collection.save_csv()
+
+
+# If empty start-date set by user then no date should be set.
+# Otherwise set user-specified date or use current date
+def start_date(entry):
+    if "start-date" in entry:
```

Comment:
we can reduce indentation here by rewriting the first check as a guard...

`if "start-date" not in entry:
        return ""`